### ISSUE TITLE
Calculations in WGS 84 are inaccurate 

### ISSUE THREAD

**Issue by kadyb** [bczernecki/climate#60]
I think you should consider saving the location of IMGW stations in EPSG 2180 coordinate system as default. Currently distance calculations based on WGS 84 are inaccurate (several km difference).

```r
library("sf")
library("climate")

data(imgw_hydro_stations)
imgw_hydro_stations = imgw_hydro_stations[imgw_hydro_stations$id == 154170050, ]
coords = c(imgw_hydro_stations$X, imgw_hydro_stations$Y)
st = nearest_stations_imgw(type = "meteo", rank = "synop", point = coords,
                           no_of_stations = 1, add_map = FALSE)
st$`distance [km]`
#> 35.601

### EPSG 2180
P1 = st_sfc(st_point(coords), crs = 4326)
P1 = st_transform(P1, crs = 2180)
P2 = st_sfc(st_point(c(st$X, st$Y)), crs = 4326)
P2 = st_transform(P2, crs = 2180)

as.vector(st_distance(P1, P2)) / 1000
#> 21.01992
```

**Comment by bczernecki** [comment]
Thanks Krzysztof for the valuable comment! 
I was considering to use GIS standards in the package, but since sf and sp are based on quite heavy system dependencies (which are causing sometimes lot of problems with installation) it was decided to use as simple approach as possible for the price of lower accurracy. 
Please also note that we have `nearest_stations_*` family of functions where EPSG 2180 might be not the best choice. 
To sum it up I  think it might be worthy adding following items:

- [ ] extra argument for `nearest_stations*` (e.g. `use_sf = TRUE`) to calculate distances based on CRS projections if `sf` package is available;
- [ ]  if there's no sf packages - use the default computing engine, but provide warning message about possible inaccuracies

**Comment by Nowosad** [comment]
Is not there a 3rd choice - just to use a better approximation of the distances? If I understand the underlining code, it uses the Euclidean distance - https://github.com/bczernecki/climate/blob/1c23d1f786d01cf8f2694c726365ecf51e7bf47d/R/nearest_stations_imgw.R#L72.
However, we could use https://pl.wikibooks.org/wiki/Astronomiczne_podstawy_geografii/Odleg%C5%82o%C5%9Bci - it would require implementing a new dist function.

**Comment by kadyb** [comment]
Here I implemented formulae to calculate the distance between points on a sphere. The simple formula is the first formula from the source from Jakub. Two more (haversine and Vincenty) come from [here](https://en.wikipedia.org/wiki/Great-circle_distance#Computational_formulas). I haven't done any accuracy and speed benchmarks, but referring to Wikipedia, the Vincenty formula is slower, but more accurate.

```r
spherical_dist = function(p1, p2, formula = NULL) {

  if (is.null(formula))
    stop("formula should be: 'simple', 'haversine' or 'Vincenty'")

  if (p1[1] > 180 || p2[1] > 180) stop("x should be longitude")
  if (p1[2] > 90 || p2[2] > 90) stop("y should be latitude")

  r = 6371009 # mean earth radius in meters
  deg2rad = function(deg) {deg * pi / 180}

  vec = c(p1, p2)
  vec = sapply(vec, FUN = deg2rad)
  diff_long = vec[3] - vec[1]
  diff_lat = vec[4] - vec[2]

  if (formula == "simple") {
    d = sqrt(diff_lat^2 + (cos((vec[2] + vec[4]) / 2) * diff_long)^2)
  }

  if (formula == "haversine") {
    d = (sin(diff_lat / 2))^2 + (1 - (sin(diff_lat / 2))^2 - (sin((vec[2] + vec[4]) / 2))^2) * (sin(diff_long / 2))^2
    d = 2 * asin(sqrt(d))
  }

  if (formula == "Vincenty") {
    num = (cos(vec[4]) * sin(diff_long))^2 + (cos(vec[2]) * sin(vec[4]) - sin(vec[2]) * cos(vec[4]) * cos(diff_long))^2
    denom = sin(vec[2]) * sin(vec[4]) + cos(vec[2]) * cos(vec[4]) * cos(diff_long)
    d = atan(sqrt(num) / denom)
  }

  d = d * r
  return(d/1000) # output in km

}


# correct: 105.4737 km
p1 = c(18.633333, 54.366667) # Gdańsk
p2 = c(17.016667, 54.466667) # Słupsk
spherical_dist(p1, p2, formula = "simple") #> 105.1923
spherical_dist(p1, p2, formula = "haversine") #> 105.19
spherical_dist(p1, p2, formula = "Vincenty") #> 105.19

# correct: 21.01992 km
p1 = c(17.16716, 54.64006) # GARDNA WIELKA
p2 = c(16.85410, 54.58831) # USTKA
spherical_dist(p1, p2, formula = "simple") #> 20.96341
spherical_dist(p1, p2, formula = "haversine") #> 20.96339
spherical_dist(p1, p2, formula = "Vincenty") #> 20.96339
```

_Correct distances in comments are calculated in the planar coordinate system (EPSG 2180), but there are also some distortions._
